src/lib/zkapps_examples/add_events/zkapps_add_events.ml

```diff
@@ -0,0 +1,65 @@
+open Core_kernel
+open Snark_params.Tick.Run
+open Signature_lib
+open Mina_base
+open Zkapps_examples
+
+let initialize public_key =
+  Zkapps_examples.wrap_main (fun () ->
+      Party_under_construction.In_circuit.create
+        ~public_key:(Public_key.Compressed.var_of_t public_key)
+        ~token_id:Token_id.(Checked.constant default)
+        () )
+
+type _ Snarky_backendless.Request.t +=
+  | Updated_events : Field.Constant.t array list Snarky_backendless.Request.t
+
+let update_events_handler (updated_events : Field.Constant.t array list)
+    (Snarky_backendless.Request.With { request; respond }) =
+  match request with
+  | Updated_events ->
+      respond (Provide updated_events)
+  | _ ->
+      respond Unhandled
+
+let events =
+  List.init 4 ~f:(fun n -> Array.init (n + 1) ~f:(fun _ -> Field.Constant.one))
+
+(* big list of events; we'll send the pieces in 3 different Partys, then reassemble them *)
+let all_events = events @ events @ events
+
+(* Merkle hash of all the events; we'll want to get this same hash after re-assembling the events *)
+let all_events_hash = Mina_base.Zkapp_account.Events.hash all_events
+
+let update_events public_key =
+  Zkapps_examples.wrap_main (fun () ->
+      let party =
+        Party_under_construction.In_circuit.create
+          ~public_key:(Public_key.Compressed.var_of_t public_key)
+          ~token_id:Token_id.(Checked.constant default)
+          ()
+      in
+      Party_under_construction.In_circuit.set_events events party )
```

Comment:
Request the events that you are passing using `Update_events` handler? otherwise you are hardcoding it in the circuit.